domain.go

```diff
@@ -0,0 +1,25 @@
+package clerk
```

Comment:
❓ Couldn't we use the `domain` package here instead?

Reply:
We could, but I'm afraid that at some point we might need this type from another package. Imagine an Instance which has a nested Domain.

I believe it's safer to have these types for "models" in the `clerk` package so we don't run into any circular dependecy problem in the future.

Reply:
I see, I thought that we might want to reuse structs. Do you think that if we introduced a `models/<service>` kind of structure, we might run into circular dependency problems? Just thinking out loud here:

```go
import (
  domainmodels "github.com/clerk/clerk-sdk-go/v2/models/domains"
)
```

Reply:
Thanks for the suggestion! Your solution would certainly avoid circular dependency, but isn't it a bit too much packaging?

Is there any reason why using the clerk package might be problematic?

Reply:
It's not problematic, it's only that packages provide a native namespace and the import aliases provide some more naming flexibility for someone that uses them. Names also become more uniform.

```go
package domain

// Debatable name :)
type Object struct{}
type List struct {
  Data []Object `json:"data"`
}
```

```go
import (
  domainmodels "github.com/clerk/clerk-sdk-go/v2/models/domains"
) 

func HandleDomains() {
  var domainList domainmodels.List
  //...
}
```

Just an idea here, not insisting on this, it might be overly verbose/complicated for some reason. Noting now, since it would be a breaking change afterwards!